### PR TITLE
Add mps to testing init and fixing test cases

### PR DESCRIPTION
**Pull request by NripeshN**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by NripeshN** [comment]
Hello Tutorials are failing,
Please approve https://github.com/kornia/tutorials/pull/62 for tutorials to pass.

File: test/geometry/test_homography.py [hunk 1]
```diff
@@ -392,12 +394,15 @@ def test_smoke(self, device, dtype):
...
     @pytest.mark.parametrize("batch_size, num_points", [(1, 4), (2, 5), (3, 6)])
     def test_shape(self, batch_size, num_points, device, dtype):
-        B, N = batch_size, num_points
-        points1 = torch.rand(B, N, 2, device=device, dtype=dtype)
-        points2 = torch.rand(B, N, 2, device=device, dtype=dtype)
-        weights = torch.ones(B, N, device=device, dtype=dtype)
-        H = find_homography_dlt_iterated(points1, points2, weights, 5)
-        assert H.shape == (B, 3, 3)
+        if sys.platform == 'darwin' and torch.__version__ == '1.9.1':
+            pytest.skip("Known bug in torch 1.9.1 on macos")
```
Comment (edgarriba) [inline test/geometry/test_homography.py:398]: i suggest to use the following pattern
https://github.com/kornia/kornia/blob/e432b7b2e13eb48602fd361947ef9119988920a3/test/tracking/test_planar_tracking.py#L34

torch.__version__ not always gives a constant return pattern

**Review by edgarriba — CHANGES_REQUESTED** [review]

File: test/geometry/test_homography.py (hunk 1 above)
Comment (NripeshN) [reply test/geometry/test_homography.py:397]: I have rewritten the code like this as I want it to only skip if it's mac. Is that okay?
if sys.platform == 'darwin' and torch_version_le(1, 9, 1):
            pytest.skip("Known bug in torch 1.9.1 on macos")

File: test/geometry/test_homography.py [hunk 2]
```diff
@@ -392,12 +395,15 @@ def test_smoke(self, device, dtype):
 
     @pytest.mark.parametrize("batch_size, num_points", [(1, 4), (2, 5), (3, 6)])
```
Comment (edgarriba) [inline test/geometry/test_homography.py:396]:
```suggestion
    @pytest.mark.parametrize("batch_size, num_points", [(1, 4), (2, 5), (3, 6)])
    @pytest.mark.skipif(sys.platform == "darwin" and torch_version_le(1, 9, 1), reason="Known bug in torch 1.9.1 on macos")
```

**Review by edgarriba — APPROVED** [review]

**Pull request merged**